Mathlib/MeasureTheory/Integral/Periodic.lean

```diff
@@ -221,14 +225,55 @@ protected theorem intervalIntegral_preimage (t : ℝ) (f : UnitAddCircle → E)
 
 end UnitAddCircle
 
-variable {E : Type*} [NormedAddCommGroup E] [NormedSpace ℝ E]
-
+/-!
+## Interval integrability of periodic functions
+-/
 namespace Function
 
 namespace Periodic
 
+variable {E : Type*} [NormedAddCommGroup E]
+
 variable {f : ℝ → E} {T : ℝ}
 
+/-- A periodic function is interval integrable over every interval if it is interval integrable
+over one period. -/
+theorem intervalIntegrable {t : ℝ} (h₁f : Function.Periodic f T) (hT : 0 < T)
+  (h₂f : IntervalIntegrable f MeasureTheory.volume t (t + T)) (a₁ a₂ : ℝ) :
+    IntervalIntegrable f MeasureTheory.volume a₁ a₂ := by
+  -- Replace [a₁, a₂] by [t - n₁ * T, t + n₂ * T], where n₁ and n₂ are natural numbers
+  obtain ⟨n₁, hn₁⟩ := exists_nat_ge ((t - min a₁ a₂) / T)
+  obtain ⟨n₂, hn₂⟩ := exists_nat_ge ((max a₁ a₂ - t) / T)
+  have : Set.uIcc a₁ a₂ ⊆ Set.uIcc (t - n₁ * T) (t + n₂ * T) := by
+    rw [Set.uIcc_subset_uIcc_iff_le]
+    constructor
+    · calc min (t - n₁ * T) (t + n₂ * T)
+      _ ≤ (t - n₁ * T) := by apply min_le_left
+      _ ≤ min a₁ a₂ := by linarith [(div_le_iff₀ hT).1 hn₁]
+    · calc max a₁ a₂
+      _ ≤ t + n₂ * T := by linarith [(div_le_iff₀ hT).1 hn₂]
+      _ ≤ max (t - n₁ * T) (t + n₂ * T) := by apply le_max_right
+  apply IntervalIntegrable.mono_set _ this
+
```

Comment:
```suggestion
```
In general, we avoid empty lines inside proofs, as this helps noticing when proofs are over and separating theorems cleanly. Adding comments like you do is enough to structure the proof. If you really want to have an empty line, you can put an empty comment like `  --`.

Reply:
Roger.